### PR TITLE
Add API Key support

### PR DESCRIPTION
**Pull request by cristianchaparroa**
## JIRA Ticket URL
https://merlinapp.atlassian.net/browse/JAC-75

## Description
- Add API key support suing the header `Api-Key`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I did add reviewers to this PR
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I created the unit tests and they are passing

### CODE REVIEW

File: authorizer_test.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	}
+
+	for _, tc := range test {
+		performHTTPTest(t, tc.ServerApiKey, tc.RequestApiKey,tc.CodeExpected, tc.ShouldCalled)
+	}
+}
+
+func performHTTPTest(t *testing.T, serverApiKey, requestApiKey string, expectedCode int, shouldCalled bool) {
+
+	//called := false
```
Comment (daltemen) [inline authorizer_test.go]: please delete it

File: authorizer_test.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+	req := httptest.NewRequest("GET", "/", nil)
+
+	req.Header.Set(ApiKey, requestApiKey)
+	w := httptest.NewRecorder()
+	httHandler.ServeHTTP(w, req)
+
+	resp := w.Result()
+
+	//assert.Equal(t, shouldCalled, called)
```
Comment (daltemen) [inline authorizer_test.go]: the same here

**Review by oscanezing — APPROVED** [review]

File: authorizer_test.go (hunk 1 above)
Comment (cristianchaparroa) [reply authorizer_test.go]: Done

File: authorizer_test.go (hunk 2 above)
Comment (cristianchaparroa) [reply authorizer_test.go]: Done

**Review by E-ricus — APPROVED** [review]

**Review by daltemen — APPROVED** [review]

**Pull request merged**